scripts/marketing_data_test.py

```diff
@@ -0,0 +1,481 @@
+        noncausal_est = XGBClassAUCScore(**y_params)
+        # TODO: Confirm with Carl that redefining treat_est_combo is fine since we removed
+        # collinear features.
+        treat_est_combo = pd.concat([X,T], axis=1)
+        noncausal_est.fit(treat_est_combo, y)
+        logger.info("Standard model fit done!")
+        # TODO: Confirm with Carl if replacing treat_est_combo with below is correct.
+        standard_predictions = pd.DataFrame(noncausal_est.predict_proba(treat_est_combo,)[:,1])
+        standard_predictions = pd.concat([standard_predictions,treat_identifiers],axis=1)
+        logger.info(f"Non-causal model fit is done!")
+        logger.info(f'{pd.DataFrame(standard_predictions).describe()}')
+
+        for t_param_combo in product(*t_cv_param_dict.values()):
+            t_params = dict(zip(t_cv_param_dict.keys(), t_param_combo))
+
+            for dml_param_combo in product(*dml_cv_param_dict.values()):
+                # Map the combination back to the parameter names
+                dml_params = dict(zip(dml_cv_param_dict.keys(), dml_param_combo))
+
+                logger.info("---------------------------------------------------------------------")
+                logger.info(
+                    f"Starting {model_name}.fit dml_params = {dml_params}"
+                    f", Y params = {y_params}, T params = {t_params}"
+                )
+
+                for parameter in dml_param_combo:
+                    output_results.write(f"{parameter},")
+                for parameter in y_param_combo:
+                    output_results.write(f"{parameter},")
+                for parameter in t_param_combo:
+                    output_results.write(f"{parameter},")
+
+                if model_name == "SparseLinearDML":
+                    est = SparseLinearDML(
+                        model_t=XGBRegR2Score(**t_params),
+                        model_y=XGBClassAUCScore(**y_params),
+                        discrete_outcome=True,
+                        diagnostic_level=2,
+                        cv=cv_folds,
+                        **dml_params,
+                    )
+                elif model_name == "CausalForestDML":
+                    est = CausalForestDML(
+                        model_t=XGBRegR2Score(**t_params),
+                        model_y=XGBClassAUCScore(**y_params),
+                        discrete_outcome=True,
+                        cv=cv_folds,
+                        **dml_params,
+                    )
+                else:
+                    raise ValueError(f"Ba")
+
+                est.fit(ytrain, T=Ttrain, X=Xtrain, W=None)
+
+                for fold_score in est.nuisance_scores_y[0]:
+                    output_results.write(f"{fold_score},")
+                for fold_scores in est.nuisance_scores_t[0]:
+                    for score in fold_scores:
+                        output_results.write(f"{score},")
```

Comment:
I'm still trying to understand why we want to save `nuisance_scores_t` for each treatment. For example, we can directly get a summary of mean point estimate using `summary()`.

<img width="1022" alt="image" src="https://github.com/user-attachments/assets/b107b7fe-61f9-4af7-b177-d87e41caef52">